kernel/src/process.rs

```diff
@@ -96,6 +96,19 @@ impl fmt::Debug for ProcessId {
     }
 }
 
+/// The read permissions the process has
+/// If the permissions exist this will be: `Some((num_ids, ids))`
+/// Where num_ids is the number of `read_ids` and `ids` is an array of the
+/// `read_ids`.
+pub type ReadPermissions = Option<(usize, [u32; 8])>;
```

Comment:
This still puts `8` in the public capsule-facing process API. I'm hesitant to go that way.

Would you be ok with a `persistent_storage_has_write_access(storage_id: u32) -> bool`-style API?

Reply:
That works fine for checking the permissions, but we also need to store the permissions in flash when we write data. So we need a way to get them to then write them.

Also we check permissions in `KVStore` as this allows either applications or the kernel to store data. At this point though we don't have a `Process` as the kernel might be using it. So there isn't really a way we could check against a `Process` here. Instead `KVDriver` get the app permissions and passes them down. This way the kernel can directly use `KVStore` if it wants